maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java

```diff
@@ -266,5 +279,41 @@ private String getMavenVersion()
 
         return props.getProperty( "version", "unknown-version" );
     }
+    
+    private Collection<FileTransformer> getTransformersForArtifact( final org.eclipse.aether.artifact.Artifact artifact,
+                                                                    final SessionData sessionData )
+    {
+        TransformerContext context = (TransformerContext) sessionData.get( TransformerContext.KEY );
+        Collection<FileTransformer> transformers = new ArrayList<>();
+        
+        // In case of install:install-file there's no transformer context, as the goal is unrelated to the lifecycle. 
+        if ( "pom".equals( artifact.getExtension() ) && context != null )
+        {
+            transformers.add( new FileTransformer()
+            {
+                @Override
+                public InputStream transformData( File pomFile )
+                    throws IOException, TransformException
+                {
+                    try
+                    {
+                        return new ConsumerModelSourceTransformer().transform( pomFile.toPath(), context );
+                    }
+                    catch ( org.apache.maven.model.building.TransformerException e )
```

Comment:
Could this type be imported? And `Artifact` at ll. 309-310.

Reply:
yes, now it can, it used to have collissions with `javax.xml.transform.TransformerException` and `org.apache.maven.artifact.Artifact`